File: backend/src/main/java/com/festago/artist/dto/ArtistResponse.java
package com.festago.artist.dto;

import com.festago.artist.domain.Artist;

public record ArtistResponse(
    Long id,
    String name,
    String profileImage
) {

    public static ArtistResponse from(Artist artist) {
        return new ArtistResponse(artist.getId(), artist.getName(), artist.getProfileImage());
    }
}

File: android/festago/data/src/main/java/com/festago/festago/data/dto/artist/ArtistResponse.kt
package com.festago.festago.data.dto.artist

import com.festago.festago.domain.model.artist.Artist
import kotlinx.serialization.Serializable

@Serializable
data class ArtistResponse(
    val id: Long,
    val name: String,
    val profileImageUrl: String,
) {
    fun toDomain() = Artist(
        id = id,
        name = name,
        imageUrl = profileImageUrl,
    )
}
